### PR TITLE
Valet Theme Updater — Token Transformer

### PR DESCRIPTION
**Pull request by djalmaaraujo**
## Description

Introducing a way to have the VIP Design System (Figma files) integrated with all our React components 🥳 . The long explanation is described in the README.md file changes in this Pull Request.

The long history short is: We now can read all theming information we need from a tokens JSON file and parse it so the React theme can be read and used in the components system.

New npm script command added:

```bash
npm run theme-update
```

## Checklist

- [ ] This PR has good automated test coverage
- [ ] The storybook for the component has been updated

## Steps to Test

1. Pull down PR.
2. `npm run theme-update`.
3. Check the `src/generated/valet-theme.json` file
4. Run `npm run dev`
5. See the new colors are applied to all components

### CODE REVIEW

File: src/system/theme/index.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ const outline = {
 	outlineStyle: 'solid',
 	outlineColor: '#ffffff',
 	outlineWidth: '1px',
-	boxShadow: `0 0 0 1px #fff, 0 0 0 3px ${ getColor( 'focus' ) }`,
+	boxShadow: `0 0 0 1px ${ getColor( 'focus', 'inset' ) }, 0 0 0 3px ${ getColor( 'focus' ) }`,
```
Comment (luiztiago) [inline src/system/theme/index.js:59]: 😍

**Review by brunobasto — APPROVED** [review]
This is awesome 💯

**Comment by luiztiago** [comment]
LGTM :shipit:

**Review by luiztiago — APPROVED** [review]

**Review by ingeniumed — APPROVED** [review]
Amazing!

**Review by alecgeatches — APPROVED** [review]
Awesome! It's cool that you can reuse color references in the design system and can drop in the tokens as-is.

Are you planning to integrate other tokens like typography in here from the design system as well?

**Comment by djalmaaraujo** [comment]
> Awesome! It's cool that you can reuse color references in the design system and can drop in the tokens as-is.
> 
> Are you planning to integrate other tokens like typography in here from the design system as well?

Yes, that's the idea for future improvements @alecgeatches 🥳

**Pull request merged**